### PR TITLE
Aftershock: Jet packs

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary
Mods "Aftershock: Add functional jet packs"

#### Purpose of change

It is the future. Where is my jet pack?
And also I just re watched the Star Wars movies a few days ago.

Inspired by the new jet pack exosuit module. I just thought it'd be cool if it could actually let you do some rocket jumping.

#### Describe the solution
Jet packs still work as a Spell powered dash, with the addition that the dash will also grant a few instants of levitation, used to either simulate a brief instant of powered flight or to allow actions while the player  is at the cusp of the parabolic jump. 
Additionally a jet pack might be used to break a fall from any distance, with this ability costing the same fuel as a regular activation cost.

Also included are a profession that starts with a new jet pack item and individual spawns for the pack.

Some other changes that were done to have this happen:

-Levitation effect: Grants you the ability to walk in open air, but not the ability to go up or down. This is used to simulate a few instants of controlled flight at the cusp of the jump. Or to let you hover aim a gun/throw.

-EXTRA_EFFECTS_FIRST Spell flag: Casts the spell secondary spells before the principal effect. In this case its used to give you  the Levitation flag before the jump, so you don't instantly fall down if you target open air. Although I foresee it can have other uses, primarily with teleport spells.

#### Describe alternatives you've considered

No jet pack no future

#### Testing

Spawned as the new profession, flew around several times.

#### Additional context

Boba Fett?! Where??

### CODE REVIEW

**Comment by Venera3** [comment]
Does the gravity check break `WALL_CLING`ing up sheer walls? I believe there's already an AS armor utilizing that flag.

**Comment by John-Candlebury** [comment]
Made it so you wont fall if you have that flag + an adjacent fall, since I supposed its meant to work as drill in climbing gear. Thats how AFS *and its ounly implementation, is using it at least,

File: src/trapfunc.cpp [hunk 1]
```diff
@@ -1231,7 +1249,18 @@ bool trapfunc::ledge( const tripoint &p, Creature *c, item * )
                                     _( "<npcname> flaps their wings and flutters down gracefully." ) );
     } else if( you->has_active_bionic( bio_shock_absorber ) ) {
         you->add_msg_if_player( m_info,
-                                _( "You hit the ground hard, but your grav chute handles the impact admirably!" ) );
+                                _( "You use your " ) );
```
Comment (BevapDin) [inline src/trapfunc.cpp:1252]: This message makes no sense. What are you using here?

Comment (John-Candlebury) [reply src/trapfunc.cpp]: Thanks, probably a mistake from when I squashed the original commits

**Pull request merged**